src/main/java/edu/hm/hafner/analysis/NullSafeListInheritance.java

```diff
@@ -0,0 +1,68 @@
+package edu.hm.hafner.analysis;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Implementierung einer Null-Safe List mit Vererbung.
+ *
+ * Vorteil:
+ * - es muessen nicht alle Methoden neu definiert werden
+ *
+ * Nachteil:
+ * -  Verletzung des Liskovschen Substitutionsprinzips
```

Comment:
Wieso? Die Schnittstelle lässt das doch zu?

Reply:
Okay, dann habe Ich da etwas falsch verstanden. Wichtig ist also das List Interface, welches diese Vererbung Version über ArrayList implementiert? Weil da steht, das es auch Listen geben kann, die null nicht akzeptieren? 

Ich habe gedacht, da man jetzt ja auch folgendes schreiben kann: 
```ArrayList<Integer> list = new NullSafeListInheritance<>();```
und laut [JavaDoc](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html) bei ArrayList folgendes gilt: "permits all elements, including null" das diese Version deshalb das LSP verletzt?!

Reply:
Ah, in dieser Richtung haben Sie Recht. Aber darauf würde ich nie kommen, da ich immer schreiben würde 

```
List<Integer> list = new NullSafeListInheritance<>();
list = new ArrrayList<>();
list = new NullSafeList<>()
```